File: Data Structures & Algorithms/find-lucky-integer-in-an-array/submission-0.cpp
class Solution {
public:
    int findLucky(vector<int>& arr) {
        map<int,int>m;
        for(int i=0;i<arr.size();i++){
            m[arr[i]]++;
        }
        int maxi=-1;
        for(auto x:m){
            if(x.first==x.second){
                maxi=max(maxi,x.first);
            }
        }
        return maxi;
    }
};

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        m=set()
        r=l=0
        maxi=0
        while r<len(s):
            if s[r] not in m:
                m.add(s[r])
            else:
                l=r
            r+=1
            maxi=max(maxi,r-l+1)
        return maxi-1

